setup.py

```diff
@@ -2183,31 +2171,48 @@ def detect_ctypes(self):
                                sources=['_ctypes/_ctypes_test.c'],
                                libraries=['m']))
 
+        ffi_inc = sysconfig.get_config_var("LIBFFI_INCLUDEDIR")
+        ffi_lib = None
+
         ffi_inc_dirs = self.inc_dirs.copy()
         if MACOS:
-            if '--with-system-ffi' not in sysconfig.get_config_var("CONFIG_ARGS"):
-                return
-            # OS X 10.5 comes with libffi.dylib; the include files are
-            # in /usr/include/ffi
-            ffi_inc_dirs.append('/usr/include/ffi')
-
-        ffi_inc = [sysconfig.get_config_var("LIBFFI_INCLUDEDIR")]
-        if not ffi_inc or ffi_inc[0] == '':
-            ffi_inc = find_file('ffi.h', [], ffi_inc_dirs)
-        if ffi_inc is not None:
-            ffi_h = ffi_inc[0] + '/ffi.h'
+            # XXX: The define should only be added when actually using the system
```

Comment:
the comment is correct.  we don't want `__builltin_available` checks for libffi functionality unless the libffi being used is the one bundled with the OS

Reply:
I agree. There's another bug as well: the code will overwrite ffi_inc if it was set earlier (user has pkg-config installed and a non-system version of libffi, for example in homebrew).  The define should only be set when we're using ffi_in_sdk and that directory exists. 

I'll push an updated version shortly.